src/main/java/teammates/common/util/Version.java

```diff
@@ -0,0 +1,167 @@
+package teammates.common.util;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+import com.google.appengine.api.modules.ModulesService;
+import com.google.appengine.api.modules.ModulesServiceFactory;
+
+public class Version {
+    private static ModulesService modulesService = ModulesServiceFactory.getModulesService();
+    private Integer major;  //  the first number
+    private Integer minor;  //  the second number
+    private String patch;   //  the rest of version
+    
+    /**
+     * Creates a new instance of Version from string.
+     * It accepts either XX-XX-XXXXX or XX.XX.XXXX format.
+     */
+    public Version(String versionInString) {
+        String[] list;
+        if (versionInString.contains("-")) {
+            list = versionInString.split("-", 3);   // split into at most 3 parts
+        } else {
+            list = versionInString.split(".", 3);
+        }
+        if (list.length > 0) {
+            major = Integer.parseInt(list[0]);
+        }
+        if (list.length > 1) {
+            minor = Integer.parseInt(list[1]);
+        }
+        if (list.length > 2) {
+            patch = list[2];
+        }
+    }
+    
+    /**
+     * Compares by string representation.
+     */
+    public boolean equals(Object anotherVersion) {
+        return toString().equals(anotherVersion.toString());
+    }
```

Comment:
Just a practice in Java, if you override `equals` you would've to override `hashCode` as well so consider doing that.